src/Microsoft.ServiceFabric.Actors.Migration/MigrationOrchestrator.cs

```diff
@@ -52,168 +52,189 @@ public async Task StartMigration(CancellationToken cancellationToken)
                     cancellationToken.ThrowIfCancellationRequested();
-                    using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+                    await this.StartDowntimePhaseWorkload(lastAppliedSNInCatchupPhase, cancellationToken);
+                }
+                else if (phase == MigrationPhase.Downtime)
                     {
-                        workerCountValue = await this.metadataDict.TryGetValueAsync(tx, MigrationConstants.CopyWorkerCountKey);
-                        endSNValue = await this.metadataDict.TryGetValueAsync(tx, MigrationConstants.CopyPhaseEndSNKey);
+                    await this.ResumeDowntimePhaseFromFailover(cancellationToken);
                     }
+                else if (phase == MigrationPhase.Uninitialized)
+                {
+                    await this.StartCompleteMigration(cancellationToken);
+                }
+            }
+            else
+            {
+                await this.StartCompleteMigration(cancellationToken);
+            }
+        }
 
-                    int.TryParse(Encoding.ASCII.GetString(workerCountValue.Value), out var workerCount);
-                    long.TryParse(Encoding.ASCII.GetString(endSNValue.Value), out var endSN);
-                    var isWorkerTaskIncomplete = new List<bool>(workerCount);
+        private async Task<long> ResumeCopyPhaseFromFailover(CancellationToken cancellationToken)
+        {
+            ActorTrace.Source.WriteInfo("MigrationOrchestrator", "Resuming migration from copy phase failover");
+            ConditionalValue<byte[]> copyPhaseWorkerCountValue, copyPhaseEndSNValue;
+            cancellationToken.ThrowIfCancellationRequested();
+            using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+            {
+                copyPhaseWorkerCountValue = await this.metadataDict.TryGetValueAsync(tx, MigrationConstants.CopyWorkerCountKey);
+                copyPhaseEndSNValue = await this.metadataDict.TryGetValueAsync(tx, MigrationConstants.CopyPhaseEndSNKey);
+            }
 
-                    cancellationToken.ThrowIfCancellationRequested();
-                    using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
-                    {
-                        for (int i = 0; i < workerCount; i++)
-                        {
-                            var workerStatusValue = await this.metadataDict.TryGetValueAsync(tx, MigrationConstants.GetCopyWorkerStatusKey(i));
-                            MigrationStatus.TryParse(Encoding.ASCII.GetString(workerStatusValue.Value), out MigrationStatus status);
-                            isWorkerTaskIncomplete[i] = status != MigrationStatus.Completed;
-                        }
-                    }
+            int.TryParse(Encoding.ASCII.GetString(copyPhaseWorkerCountValue.Value), out var workerCountForCopyPhase);
```

Comment:
What happens if TryParse fails ? If we are using TryParse/TryGetValueAsync, we need to make sure we are handling TryParse returning false and TryGetValueAsync returning ConditionalValue with null value. 
Or you can simply call Parse and let the service fail with framework specific error.